rust/headless-client/src/windows.rs

```diff
@@ -0,0 +1,489 @@
+    command: CmdIpc,
+}
+
+#[derive(clap::Subcommand)]
+enum CmdIpc {
+    #[command(hide = true)]
+    DebugIpcService,
+    IpcService,
+}
+
+impl Default for CmdIpc {
+    fn default() -> Self {
+        Self::IpcService
+    }
+}
+
+// The return value is useful on Linux
+#[allow(clippy::unnecessary_wraps)]
+pub(crate) fn check_token_permissions(_path: &Path) -> Result<()> {
+    // TODO: Make sure the token is only readable by admin / our service user on Windows
+    Ok(())
+}
+
+pub(crate) fn default_token_path() -> std::path::PathBuf {
+    // TODO: System-wide default token path for Windows
+    PathBuf::from("token.txt")
+}
+
+/// Only called from the GUI Client's build of the IPC service
+///
+/// On Windows, this is wrapped specially so that Windows' service controller
+/// can launch it.
+pub(crate) fn run_only_ipc_service() -> Result<()> {
+    let cli = CliIpcService::parse();
+    match cli.command {
+        CmdIpc::DebugIpcService => run_debug_ipc_service(cli),
+        CmdIpc::IpcService => windows_service::service_dispatcher::start(SERVICE_NAME, ffi_service_run).context("windows_service::service_dispatcher failed. This isn't running in an interactive terminal, right?"),
+    }
+}
+
+fn run_debug_ipc_service(cli: CliIpcService) -> Result<()> {
+    crate::debug_command_setup()?;
+    let rt = tokio::runtime::Runtime::new()?;
+    let mut ipc_service = pin!(ipc_listen(cli));
+    let mut signals = Signals::new()?;
+    rt.block_on(async {
+        std::future::poll_fn(|cx| {
+            match signals.poll(cx) {
+                Poll::Ready(SignalKind::Hangup) => {
+                    return Poll::Ready(Err(anyhow::anyhow!(
+                        "Impossible, we don't catch Hangup on Windows"
+                    )));
+                }
+                Poll::Ready(SignalKind::Interrupt) => {
+                    tracing::info!("Caught Interrupt signal");
+                    return Poll::Ready(Ok(()));
+                }
+                Poll::Pending => {}
+            }
+
+            match ipc_service.as_mut().poll(cx) {
+                Poll::Ready(Ok(())) => {
+                    return Poll::Ready(Err(anyhow::anyhow!(
+                        "Impossible, ipc_listencan't return Ok"
+                    )));
+                }
+                Poll::Ready(Err(error)) => {
+                    return Poll::Ready(Err(error).context("ipc_listen failed"));
+                }
+                Poll::Pending => {}
+            }
+
+            Poll::Pending
+        })
+        .await
+    })
+}
+
+// Generates `ffi_service_run` from `service_run`
+windows_service::define_windows_service!(ffi_service_run, windows_service_run);
+
+fn windows_service_run(_arguments: Vec<OsString>) {
+    if let Err(_e) = fallible_windows_service_run() {
+        todo!();
+    }
+}
+
+#[cfg(debug_assertions)]
+const SERVICE_RUST_LOG: &str = "debug";
+
+#[cfg(not(debug_assertions))]
+const SERVICE_RUST_LOG: &str = "info";
```

Comment:
```suggestion
const SERVICE_RUST_LOG: &str = "str0m=warn,info";
```

Does this affect the str0m log level since it's the service process?

Reply:
Yeah. I'll cut and paste the filters from the GUI side in an upcoming commit